src/test/scala/fundamentals/level03/ExceptionExercisesTest.scala

```diff
@@ -106,15 +106,23 @@ class ExceptionExercisesTest extends FunSpec with TypeCheckedTripleEquals {
 
   describe("collectErrors") {
 
-    it("should return a List Exceptions thrown while processing inputs") {
-
-      def exceptionEq(e1: Exception, e2: Exception): Boolean =
-        e1.getClass == e2.getClass && e1.getMessage == e2.getMessage
+    def exceptionEq(e1: Exception, e2: Exception): Boolean =
+      e1.getClass == e2.getClass && e1.getMessage == e2.getMessage
+
+    val expectedErrors = List(
+      new InvalidAgeValueException("provided age is invalid: 5o"),
+      new InvalidAgeRangeException("provided age should be between 1-120: 200"),
+      new InvalidAgeRangeException("provided age should be between 1-120: 0"),
+      new EmptyNameException("provided name is empty"),
+      new InvalidAgeRangeException("provided age should be between 1-120: 1000"),
+      new EmptyNameException("provided name is empty")
+    )
+
+    it("should return all errors") {
+      collectErrors.size === 6
```

Comment:
Because zip ignores extra items present in either of the lists. So zipping a list of 5 with another one of 6 was not failing.
That's why I added the double error tuple before the last one.

Reply:
Good point. Can we add the length check into the `"should return a List Exceptions thrown while processing inputs"` test? See comment below.